src/Microsoft.VisualStudio.Editor.Razor/DefaultEditorSettingsManager.cs

```diff
@@ -12,26 +13,95 @@ namespace Microsoft.VisualStudio.Editor.Razor
     [Export(typeof(EditorSettingsManager))]
     internal class DefaultEditorSettingsManager : EditorSettingsManager
     {
-        private readonly EditorSettingsManagerInternal _editorSettingsManager;
+        private readonly List<WeakReference<EventHandler<EditorSettingsChangedEventArgs>>> _changedHandlers;
+        private readonly object SettingsAccessorLock = new object();
+        private readonly ForegroundDispatcher _foregroundDispatcher;
+        private EditorSettings _settings;
 
         [ImportingConstructor]
-        public DefaultEditorSettingsManager(VisualStudioWorkspaceAccessor workspaceAccessor)
+        public DefaultEditorSettingsManager(ForegroundDispatcher foregroundDispatcher)
         {
-            var razorLanguageServices = workspaceAccessor.Workspace.Services.GetLanguageServices(RazorLanguage.Name);
-            _editorSettingsManager = razorLanguageServices.GetRequiredService<EditorSettingsManagerInternal>();
+            _foregroundDispatcher = foregroundDispatcher;
+            _settings = EditorSettings.Default;
+            _changedHandlers = new List<WeakReference<EventHandler<EditorSettingsChangedEventArgs>>>();
         }
 
         public override event EventHandler<EditorSettingsChangedEventArgs> Changed
         {
-            add => _editorSettingsManager.Changed += value;
-            remove => _editorSettingsManager.Changed -= value;
+            add
```

Comment:
What made you decide on weakreference here? Because it's hard to know when to detatch?

Reply:
Yes